src/codegeneration/ForOfTransformer.js

```diff
@@ -45,14 +54,43 @@ export class ForOfTransformer extends TempVarTransformer {
       assignment = parseStatement `${tree.initializer} = ${result}.value;`;
     }
 
-    return parseStatement `
+    innerStatement = parseStatement `
         for (var ${iter} =
                  ${tree.collection}[
                      $traceurRuntime.toProperty(Symbol.iterator)](),
                  ${result};
-             !(${result} = ${iter}.next()).done; ) {
+             !(${normalCompletion} = (${result} = ${iter}.next()).done); ) {
           ${assignment};
           ${tree.body};
         }`;
+  
+    while (labelledStatement = labelSet.pop()) {
+      innerStatement = new LabelledStatement(labelledStatement.location,
+          labelledStatement.name, innerStatement);
+    }
+        
+    return parseStatement `
+        try {
+          var ${normalCompletion} = false;
```

Comment:
When I move `var` out of `try`, do I wrap everything in a block { ... }? Will the block be an AnonBlock?